### PR TITLE
Integrate new protection tools into api's offer & trade services

### PR DESCRIPTION
**Pull request by ghubstan**
- Injected `OfferFilter` into `CoreOffersService` and `CoreTradesService`.
The filter constrains `getoffer(s)` results to offers an api user can take, as the first part of ongoing protection tools / api integration.

- Created a new `CoreContext` singleton.
Initially, lets `Core*Services` know if the user is using the api -- `isApiUser=true` if the current thread's name is "BisqDaemonMain", or the name contains "grpc".  We cannot check `BisqDaemonMain.class.getSimpleName()` because `:core` cannot have a circular dependency on `:daemon`, but there is probably a better way to do this than depending on the thread name set in `BisqDaemonMain#configUserThread()`.  We do this anticipating future `:desktop` dependencies on the core api, and we don't pass hardcoded `isApiUser=true` params to lower level domain objects.  

- Added `@Singleton` annotation to all Core*Service classes.
      
Related to https://github.com/bisq-network/bisq/pull/5053.

PR https://github.com/bisq-network/bisq/pull/5063 should be reviewed and merged before this one.

### CODE REVIEW

**Comment by ghubstan** [comment]
I made a mistake by making the CLI method name a posix-style opt.  I will change it to be the only positional parameter in a command.

File: core/src/main/java/bisq/core/api/CoreContext.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Slf4j
+class CoreContext {
+
+    @Inject
+    public CoreContext() {
+    }
+
+    public boolean isApiUser() {
+        String threadName = currentThread().getName();
+        return threadName.equals("BisqDaemonMain") || threadName.contains("grpc");
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/api/CoreContext.java]: I find this ugly. There must be some easier and not quite as ugly way to do this.

**Review by sqrrm — COMMENTED** [review]
It hurts my soul somewhat with that thread parsing to figure out context. I think we should try to think a bit more before merging something like that.

File: core/src/main/java/bisq/core/api/CoreContext.java (hunk 1 above)
Comment (ghubstan) [reply core/src/main/java/bisq/core/api/CoreContext.java]: Resolved by commit 0e779a4

**Review by sqrrm — APPROVED** [review]
utACK

I much prefer this solution

**Pull request merged**